internal/customermanagedkeys/key_vault_or_managed_hsm_key.go

```diff
@@ -75,37 +82,37 @@ func (k *KeyVaultOrManagedHSMKey) BaseUri() string {
 	return ""
 }
 
-func parseKeyvauleID(keyRaw string, hasVersion *bool) (*parse.NestedItemId, error) {
+func parseKeyvauleID(keyRaw string, requireVersion VersionRequireness) (*parse.NestedItemId, error) {
 	keyID, err := parse.ParseOptionallyVersionedNestedKeyID(keyRaw)
 	if err != nil {
 		return nil, err
 	}
 
-	if pointer.From(hasVersion) && keyID.Version == "" {
+	if requireVersion == Versioned && keyID.Version == "" {
 		return nil, fmt.Errorf("expected a key vault versioned ID but no version information was found in: %q", keyRaw)
 	}
 
-	if hasVersion != nil && !*hasVersion && keyID.Version != "" {
+	if requireVersion == Versionless && keyID.Version != "" {
 		return nil, fmt.Errorf("expected a key vault versionless ID but version information was found in: %q", keyRaw)
 	}
 
 	return keyID, nil
 }
 
-func parseManagedHSMKey(keyRaw string, hasVersion *bool, hsmEnv environments.Api) (*hsmParse.ManagedHSMDataPlaneVersionedKeyId, *hsmParse.ManagedHSMDataPlaneVersionlessKeyId, error) {
+func parseManagedHSMKey(keyRaw string, requireVersion VersionRequireness, hsmEnv environments.Api) (*hsmParse.ManagedHSMDataPlaneVersionedKeyId, *hsmParse.ManagedHSMDataPlaneVersionlessKeyId, error) {
 	// if specified with hasVersion == True, then it has to be parsed as versionedKeyID
 	var domainSuffix *string
 	if hsmEnv != nil {
 		domainSuffix, _ = hsmEnv.DomainSuffix()
 	}
 	// versioned or optional version
-	if hasVersion == nil || pointer.From(hasVersion) {
+	if requireVersion == OptionallyVersioned || requireVersion == Versioned {
```

Comment:
Can we use a switch here to handle the 3 cases separately, to make the code more clear?